hooks/useObservable.ts

```diff
@@ -1,117 +1,97 @@
-}
+const emptyObject = {};
 
 export function useObservable<T extends IStateTreeNode>(model: T) {
-  type SnapshotType = ReturnType<typeof getSnapshot<T>>;
-  
-  const [state, setState] = useState<Record<string | symbol, unknown>>({});
-
-  useEffect(() => {
-    if (!model) return;
-
-    const disposers: ReturnType<typeof reaction>[] = [];
-
-    // Set up reaction for each property
-    Object.keys(state).forEach(prop => {
-      const disposer = reaction(
-        () => {
-          const value = model[prop as keyof T];
-          if (isMSTNode(value)) {
-            // Handle nested properties
-            const nested = {} as Record<string, unknown>;
-            Object.keys(state)
-              .filter(key => key.startsWith(`${prop}.`))
-              .forEach(key => {
-                const nestedProp = key.split('.')[1];
-                nested[nestedProp] = value[nestedProp as keyof typeof value];
-              });
-            return { value, ...nested };
-          }
-          return value;
-        },
-        (newValue) => {
-          setState(prev => ({
-            ...prev,
-            // @ts-expect-error - need to figure out some of this weird typing
-            [prop]: isMSTNode(newValue) ? newValue.value : newValue
-          }));
-        },
-        { fireImmediately: true }
-      );
-      disposers.push(disposer);
-    });
-
-    return () => disposers.forEach(d => d());
-  }, [model, Object.keys(state).join(',')]);
-
-  return new Proxy({} as SnapshotType, {
-    get: (_, property: string | symbol) => {
-      // Initialize tracking for new properties
-      if (!(property in state)) {
-        const value = model[property as keyof T];
-        
-        setState(prev => ({
-          ...prev,
-          [property]: value
-        }));
-
-        if (typeof value === 'function') {
-          return value.bind(model);
-        }
-
-        if (isMSTNode(value)) {
-          return new Proxy({} as SnapshotType, {
-            get: (_, nestedProp: string | symbol) => {
-              const nestedValue = value[nestedProp as keyof typeof value];
-              
-              if (typeof nestedValue === 'function') {
-                return nestedValue.bind(value);
-              }
-
-              // Track nested property
-              const fullProp = `${String(property)}.${String(nestedProp)}`;
-              if (!(fullProp in state)) {
-                setState(prev => ({
-                  ...prev,
-                  [fullProp]: nestedValue
-                }));
-              }
-
-              return nestedValue;
+  const [, setState] = useState(1);
+  const forceRender = useCallback(() => setState((n) => -n), []);
+
+  const pathsAccessedInRender = useRef<Set<string[]>>(new Set());
+  pathsAccessedInRender.current = new Set();
+
+  useLayoutEffect(() => {
+    const disposer = reaction(
+      function expression() {
+        //// Uncomment this to see what paths are being accessed in the render
```

Comment:
**suggestion:** let's change this to a logging function that we can configure with an optional argument to `useObservable`. Maybe just like `useObservable<T extends IStateTreeNode>(model: T, debug?: boolean)`, and then if `debug` is `true`, we run these logs?

Reply:
I've got that locally, will push up tomorrow.